### ISSUE TITLE
add filtercols and filtercols!

### ISSUE THREAD

**Issue by bkamins** [JuliaData/DataFrames.jl#2508]
We have `mapcols` so we can consider adding `filtercols` (with `view` kwarg) and `filtercols!` that would work like `filter` but would filter columns. There is no rush to add this.

**Comment by zerefwayne** [comment]
@nalimilan @bkamins This looks interesting. Can I work on this?

**Comment by bkamins** [comment]
sure - just make sure to understand the API of `filter` and `filter!` first. Also this should be synced with https://github.com/JuliaData/DataFrames.jl/pull/2417 as the API should be the same in both places.
The question is if the predicate should get a column or a "column name", column `Pair`. I think it should be just a column

**Comment by bkamins** [comment]
ah - and it should support `copycols` kwarg additionally.

**Comment by bkamins** [comment]
And the last thing - maybe wait for 0.22 release before opening a PR. Then you will be sure that you are working on a clean master (of course you can start implementation now if you want; 0.22 should be tagged this week). Finally - have a look at https://github.com/SciML/ColPrac, as following this will speed up the review process.

**Comment by zerefwayne** [comment]
I'll have a look at `filter` first. It will roughly take me a week to implement this, I'll open the PR after syncing with 0.22! Thank you for the heads-up @bkamins!

**Comment by bottine** [comment]
I don't know if that's reasonable wrt orthogonality and API, but I would have liked to have `mapcols!(fun, df, cols)` where `cols` is the usual column selector, allowing to apply a function only on specific columns. Could e.g. have `mapcols!(uppercase, df, names(df, String))` to make string values uppercase.

**Comment by bkamins** [comment]
Now the pattern for this is:
```
transform!(df, names(df, String) .=> ByRow(uppercase), renamecols=false)
```
I understand you find `mapcols!` shorter or easier to reason about?

Note that you would need to write:
```
mapcols!(col -> uppercase.(col), df, names(df, String))
```
though.

**Comment by bottine** [comment]
Ah, indeed, I didn't think it would be so short with `transform!`, you're right!
But then, 
```
mapcols!(fun, df) == transform!(df, All() .=> fun, renamecols=false)
```
so the point of `mapcols!` is mostly that it's a bit shorter?

**Comment by bkamins** [comment]
Yes, the original point is that `mapcols` was introduced much earlier in the development process of DataFrames.jl than `transform`.
